lib/data/repositories/collection_repository.dart

```diff
@@ -62,17 +73,39 @@ class CollectionRepositoryImpl implements CollectionRepository {
   }
 
   @override
-  Future<List<Stream<Collection>>> listenToAllCollections() async {
-    final rawCollections = await _db.getAll(store: _collectionStore);
-    final collectionIds = rawCollections.map((rawCollection) => rawCollection[CollectionKeys.id] as String).toList();
+  Future<List<CollectionMemos>> getCollectionMemosByIds(List<String> ids) async {
+    final collectionsFutures = ids.map((id) => _appBundle.loadJson('$_collectionsRoot/$id.json'));
+    final rawCollections = await Future.wait<dynamic>(collectionsFutures);
+
+    final castCollections = List<Map<String, dynamic>>.from(rawCollections);
+    return castCollections.map(_collectionsMemosSerializer.from).toList();
+  }
 
-    final listenFutures =
-        collectionIds.map((collectionId) => _db.listenTo(id: collectionId, store: _collectionStore)).toList();
-    final rawCollectionListeners = await Future.wait(listenFutures);
+  @override
+  Future<void> putCollectionsWithCollectionMemos(List<CollectionMemos> collections) async {
```

Comment:
This parameter shouldn't be `memos` or `collectionMemos` instead `collections`?

Reply:
That's actually confusing, but couldn't finda better naming, so no, it's a CollectionMemos instead, a list of a collection glued with a list of their respective memos metadata (this is a transient class)